src/Development/Shake/Internal/Rules/File.hs

```diff
@@ -58,6 +58,13 @@ newtype FileQ = FileQ {fromFileQ :: FileName}
 data FileA = FileA {-# UNPACK #-} !ModTime {-# UNPACK #-} !FileSize FileHash
     deriving (Typeable,Eq)
 
+-- | Result of a File rule, may contain raw file information and wether the rule did run this build
+data FileR = FileR (Maybe FileA) Bool
+    deriving (Typeable, Eq)
```

Comment:
Can you derive `Eq`? Don't you need to change it to ignore the Bool? In fact, I'm slightly surprised that it doesn't break at least one test if you look at both things...

Reply:
I didn't understand the whole code base, but I would conjecture rule results aren't used as hashmap keys and aren't usually compared for equality? What would one look up by a rule result? And the comparison in the BuiltinRule is very specialised.
Maybe that's why it didn't matter in the tests?